applications/nrf5340_audio/src/audio/streamctrl.c

```diff
@@ -423,52 +466,253 @@ static void le_audio_msg_sub_thread(void)
 			}
 
 			LOG_INF("Presentation delay %d us is set by initiator", pres_delay_us);
+
+			break;
+
+		case LE_AUDIO_EVT_PA_SYNC_LOST:
+			LOG_DBG("PA sync lost");
+
+			if (IS_ENABLED(CONFIG_BT_OBSERVER)) {
+				ret = bt_mgmt_scan_start(0, 0, BT_MGMT_SCAN_TYPE_BROADCAST, NULL);
+				if (ret) {
+					LOG_ERR("Failed to restart scanning: %d", ret);
+					break;
+				}
+
+				LOG_INF("Restarted scanning for broadcaster");
+			}
+
+			break;
+
+		case LE_AUDIO_EVT_NO_VALID_CFG:
+			LOG_WRN("No valid configurations found, will disconnect");
+
+			bt_mgmt_conn_disconnect(msg.conn, BT_HCI_ERR_REMOTE_USER_TERM_CONN);
+
 			break;
 
 		default:
-			LOG_WRN("Unexpected/unhandled event: %d", event);
+			LOG_WRN("Unexpected/unhandled le_audio event: %d", event);
+
 			break;
 		}
 
 		STACK_USAGE_PRINT("le_audio_msg_thread", &le_audio_msg_sub_thread_data);
 	}
 }
 
+/* Handle content control events */
+static void content_control_msg_sub_thread(void)
+{
+	int ret;
+
+	const struct zbus_channel *chan;
+
+	while (1) {
+		ret = zbus_sub_wait(&content_control_evt_sub, &chan, K_FOREVER);
+		ERR_CHK(ret);
+
+		struct content_control_msg msg;
+
+		ret = zbus_chan_read(chan, &msg, ZBUS_READ_TIMEOUT_MS);
+		ERR_CHK(ret);
+
+		uint8_t event = msg.event;
+
+		LOG_DBG("Received event = %d, current state = %d", event, strm_state);
+
+		switch (event) {
+		case MEDIA_PLAY:
+			ret = le_audio_play();
+			if (ret) {
+				LOG_ERR("Failed to play: %d", ret);
+			}
+
+			break;
+
+		case MEDIA_PAUSE:
+			ret = le_audio_pause();
+			if (ret) {
+				LOG_ERR("Failed to pause: %d", ret);
+			}
+
+			break;
+
+		default:
+			LOG_WRN("Unexpected/unhandled event: %d", event);
+
+			break;
+		}
+
+		STACK_USAGE_PRINT("content_control_msg_thread",
+				  &content_control_msg_sub_thread_data);
+	}
+}
+
+/**
+ * @brief	Zbus listener to receive events from bt_mgmt
+ *
+ * @note	Will in most cases be called from BT_RX context,
+ *		so there should not be too much processing done here
+ */
+static void bt_mgmt_evt_handler(const struct zbus_channel *chan)
+{
+	int ret;
+	const struct bt_mgmt_msg *msg;
+
+	msg = zbus_chan_const_msg(chan);
+	uint8_t event = msg->event;
+
+	switch (event) {
+	case BT_MGMT_CONNECTED:
+		LOG_INF("Connected");
+
+		break;
+
+	case BT_MGMT_DISCONNECTED:
+		LOG_INF("Disconnected");
+		le_audio_conn_disconnected(msg->conn);
+
+		ret = bt_content_ctrl_conn_disconnected(msg->conn);
+		if (ret) {
+			LOG_ERR("bt_content_ctrl_disconnected failed with %d", ret);
+		}
+
+		break;
+
+	case BT_MGMT_EXT_ADV_READY:
+		LOG_INF("Ext adv ready");
+		if (IS_ENABLED(CONFIG_TRANSPORT_BIS)) {
+			ret = le_audio_ext_adv_set(msg->ext_adv);
+			if (ret) {
+				LOG_WRN("Failed to set extended advertisement data");
+			}
+		}
+
+		break;
+
+	case BT_MGMT_SECURITY_CHANGED:
+		LOG_INF("Security changed");
+
+		if (CONFIG_AUDIO_DEV == GATEWAY) {
+			le_audio_conn_set(msg->conn);
+		}
+
+		ret = bt_rend_discover(msg->conn);
+		if (ret) {
```

Comment:
Should there be an EALREADY handler/functionality here as well?

Reply:
Security typically only happens once when we connect to a device. Since the services are not stored together with the bonding (I believe this is planned in host later) then we have to discover again every time we connect.